### PR TITLE
Move logger from all services to the platform package

### PR DESCRIPTION
**Pull request by ivyoverflow**
#3, #5 and #6 are closed.

### CODE REVIEW

File: book/internal/repository/mongo/cover.html [hunk 1]
```diff
@@ -0,0 +1,283 @@
+
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+		<title>mongo: Go Coverage Report</title>
```
Comment (kiryalovik) [inline book/internal/repository/mongo/cover.html]: You don't need this file to be commited.

File: book/internal/repository/postgres/migrate.go [hunk 2]
```diff
@@ -2,33 +2,32 @@
...
+	"github.com/ivyoverflow/pub-sub/book/internal/lib/types"
 )
 
-// runMigration automatically starts PostgreSQL migrations.
-func runMigration(cfg *config.PostgresConfig) error {
+// RunMigration automatically starts PostgreSQL migrations.
+func RunMigration(cfg *config.PostgresConfig) error {
 	if cfg.MigartionsPath == "" {
-		return nil
+		return types.ErrorMigrate
```
Comment (kiryalovik) [inline book/internal/repository/postgres/migrate.go:16]: Well, it's better to use for migration some cli tool, which doesn't need any code to write.

File: platform/go.sum [hunk 3]
```diff
@@ -0,0 +1,58 @@
+github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
+github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
+github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
+github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
```
Comment (kiryalovik) [inline platform/go.sum:5]: looks like you need to run `mod tidy`

**Review by kiryalovik — CHANGES_REQUESTED** [review]
If you named your PR move logger - there shouldn't be any other changes

File: README.md [hunk 4]
```diff
@@ -1 +1,46 @@
-# pub-sub
+![CircleCI Status](https://img.shields.io/circleci/build/github/ivyoverflow/pub-sub/main?style=flat-square)
+![GitHub Issues](https://img.shields.io/github/issues/ivyoverflow/pub-sub?style=flat-square)
+![GitHub Pull Requests](https://img.shields.io/github/issues-pr/ivyoverflow/pub-sub?style=flat-square)
+![GitHub Last Commit](https://img.shields.io/github/last-commit/ivyoverflow/pub-sub?style=flat-square)
+# 🦹🏻‍♀️ pub-sub
+## 🧪 What we are using?
+1. 🍃 MongoDB.
+2. 🐘 PostgreSQL.
+3. 🐹 Golang.
```
Comment (kiryalovik) [inline README.md:9]: It doesn't look like gopher xD

File: book/cmd/book/main.go [hunk 5]
```diff
@@ -29,9 +29,8 @@ func main() {
 	bookRepo := mongo.NewBookRepository(db)
 	gen := service.NewIDGenerator()
 	bookSvc := service.NewBook(bookRepo, gen)
-	bookMw := handler.NewMiddleware(ctx, log)
 	bookHandl := handler.NewBook(ctx, bookSvc, log)
-	srv := server.New(&cfg.Server, bookMw, bookHandl)
+	srv := server.New(&cfg.Server, bookHandl)
```
Comment (kiryalovik) [inline book/cmd/book/main.go:33]: don't see any logger initialization.

File: book/internal/handler/book_test.go [hunk 6]
```diff
@@ -15,11 +16,11 @@ import (
 
 	"github.com/ivyoverflow/pub-sub/book/internal/handler"
 	"github.com/ivyoverflow/pub-sub/book/internal/lib/types"
-	"github.com/ivyoverflow/pub-sub/book/internal/logger"
 	"github.com/ivyoverflow/pub-sub/book/internal/model"
 	mock_repository "github.com/ivyoverflow/pub-sub/book/internal/repository/mock"
```
Comment (kiryalovik) [inline book/internal/handler/book_test.go:20]: snail_case using isn't a Go Way

File: book/internal/handler/book_test.go [hunk 7]
```diff
@@ -345,6 +388,26 @@ func TestBookHandler_Update(t *testing.T) {
 			},
 			expectedStatusCode: 404,
 		},
+		{
+			name:           "Service error",
```
Comment (kiryalovik) [inline book/internal/handler/book_test.go:392]: What is "Service error"? What test case it checks?

File: book/internal/repository/mongo/cover.html (hunk 1 above)
Comment (ivyoverflow) [reply book/internal/repository/mongo/cover.html]: I added cover.html to the .gitgnore file. Thanks.

File: book/internal/repository/postgres/migrate.go (hunk 2 above)
Comment (ivyoverflow) [reply book/internal/repository/postgres/migrate.go:16]: Did you mean something like this?
```bash
migrate -path ./internal/repository/mongo/migrations -database mongodb://@localhost:27017/donatesdb up
```
If we are using docker containers, we will be constantly doing migrations by hand. I think this isn't the best way.
Can you explain why I should use the cli tool?

File: platform/go.sum (hunk 3 above)
Comment (ivyoverflow) [reply platform/go.sum:5]: Can I use something to automate the go mod tidy command?

File: book/cmd/book/main.go (hunk 5 above)
Comment (ivyoverflow) [reply book/cmd/book/main.go:33]: I'll fix it in the next commit. Thanks.

File: book/internal/handler/book_test.go (hunk 6 above)
Comment (ivyoverflow) [reply book/internal/handler/book_test.go:20]: I thought it had nothing to do with package names.
Should I rename it to **mockRepository**?

File: book/internal/handler/book_test.go (hunk 7 above)
Comment (ivyoverflow) [reply book/internal/handler/book_test.go:392]: I'll try to explain that in the next commit.

File: README.md (hunk 4 above)
Comment (ivyoverflow) [reply README.md:9]: Well, next time I use this emoji -> 🦍
xD

**Pull request merged**